src/workloads/scale/CursorStormMongos.yml

```diff
@@ -0,0 +1,180 @@
+  # With data distributed evenly across shards, the total size of documents to process in a
+  # given step should be equall to NumOfShards * SortStepPerShard.
+  SortStep: &SortStep
+    ^NumExpr:
+      withExpression: "num_shards * sort_step_per_shard"
+      andValues: {num_shards: *NumShards, sort_step_per_shard: *SortStepPerShard}
+  SortThreads: &SortThreads 250
+  SortRepeat: &SortRepeat 10
+
+Clients:
+  Default:
+    QueryOptions:
+      socketTimeoutMS: -1
+      maxPoolSize: 500
+    
+Actors:
+# Drop database to get rid of stale data. Useful when running locally multiple times.
+- Name: Setup
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *DBName
+        Operations:
+        - OperationName: RunCommand
+          OperationCommand: {dropDatabase: 1}
+
+# Create collection
+- Name: CreateCollection
+  Type: RunCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *DBName
+        Operations:
+        - OperationName: RunCommand
+          OperationCommand:
+            # Loader default collection name.
+            create: Collection0
+
+# Enable sharding & shard collection
+- Name: ShardCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        OnlyRunInInstance: sharded
+        Operations:
+        - OperationMetricsName: EnableSharding
+          OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *DBName
+        - OperationMetricsName: ShardCollection
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: *Namespace
+            key: {a: hashed}
+
+# Load 20,096 documents around 168KB as described by the structure in GlobalDefaults.
+- Name: LoadDocuments
+  Type: Loader
+  Threads: *LoadThreads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        CollectionCount: 1
+        Database: *DBName
+        Repeat: 1
+        Document: *Doc
+        MultipleThreadsPerCollection: true
+        DocumentCount: *NumDocs
+        BatchSize: *LoadBatchSize
+
+# Spawn many threads to test the routers capacity to handle memory pressure.
+- Name: SortMany
+  Type: RunCommand
+  Threads: *SortThreads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [4]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *SortRepeat
+        Database: *DBName
+        Operations:
+        - OperationMetricsName: SortMany
+          OperationName: RunCommand
+          OperationCommand:
+            # Loader default collection name.
+            aggregate: Collection0
```

Comment:
Can we simply stream results from the shard, e.g. a find resulting in a plain index scan or collection scan. We don't need to tax the shards, all we need is to generate a batch to feed the mongos.

Reply:
I've kept the sort based on the observation that without it, the workload takes ~2x longer to reach OOM, with ~2x more cursors being opened on the mongos. Could it be that using the sort-stage affects how data is fetched/buffered on the mongos? 

Shall we simplify the workload regardless?

Reply:
I've assumed that we might see some effects related to caching if the exact same query is run by many threads, many times. However, this seems to not be the case! Removed some noise from the workload.

> Can we simplify this workload by doing a simple unfiltered find + sort stage on an index and avoiding a blocking sort on the shards?

Updated.

> Out of curiosity, how long does it take for this stage to OOM with and without the sort stage?

It was ~40s vs ~80s. However, now its ~20s, and I could barely see any difference with or without the sort. Therefore, I've just removed the sort altogether.

Reply:
Perhaps the speed-up is due to https://jira.mongodb.org/browse/DEVPROD-4593? In any case, I'm happy without the sort stage!